From b6390568014b6ac6c6fafa54c1429fa2c8adef8f Mon Sep 17 00:00:00 2001
From: Stuart Morgan <smorgan@mythtv.org>
Date: Tue, 8 Jul 2014 22:55:22 +0100
Subject: [PATCH 2/3] MythMusic: If the 'global' POPM tag playcount was zero
 before then insert the total mythtv playcount, don't
 just increment it.

---
 mythtv/libs/libmythmetadata/metaioid3.cpp |    9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/mythtv/libs/libmythmetadata/metaioid3.cpp b/mythtv/libs/libmythmetadata/metaioid3.cpp
index 8f8f1bf..d29d762 100644
--- a/mythtv/libs/libmythmetadata/metaioid3.cpp
+++ b/mythtv/libs/libmythmetadata/metaioid3.cpp
@@ -866,11 +866,14 @@ bool MetaIOID3::writePlayCount(TagLib::ID3v2::Tag *tag, int playcount)
         popm->setCounter(playcount);
 
     // Global playcount Tag - Updated by all apps/hardware that support it
-    if (countDiff > 0)
+    PopularimeterFrame *gpopm = findPOPM(tag, "");
+    if (!gpopm)
     {
-        PopularimeterFrame *gpopm = findPOPM(tag, "");
-        gpopm->setCounter(gpopm->counter() + countDiff);
+        gpopm = new PopularimeterFrame();
+        tag->addFrame(gpopm);
+        gpopm->setEmail("");
     }
+    gpopm->setCounter(gpopm->counter() > 0 ? gpopm->counter() + countDiff : playcount);
 
     return true;
 }
-- 
1.7.10.2

